s3plugin/s3plugin.go

```diff
@@ -255,6 +256,10 @@ func readConfigAndStartSession(c *cli.Context) (*PluginConfig, *session.Session,
 	if err != nil {
 		return nil, nil, err
 	}
+
+	if config.Options.Endpoint != "" && net.ParseIP(config.Options.Endpoint) == nil {
+		sess.Handlers.Build.PushFront(removeBucketFromPath)
```

Comment:
This change breaks work with S3-compatible storages (e.g. Yandex Object Storage):

```
20231208:11:20:27 gpbackup:givi:samsung-ivanov:396025-[CRITICAL]:-exit status 1: 20231208:11:20:27 gpbackup_s3_plugin:givi:samsung-ivanov:396382-[ERROR]:-NoSuchBucket: The specified bucket does not exist.
        status code: 404, request id: b33f5d6b7077be48, host id:
```